drivers/posixaio-backend.c

```diff
@@ -486,13 +485,13 @@ posixaio_backend_init_queue(tqueue *pqueue, int size,
 		goto fail;
 	}
 
-	queue->aiocbList = calloc(size, sizeof(struct aiocb));
+	queue->aiocbList = calloc(size, sizeof(struct aiocb*));
```

Comment:
Would `calloc(size, sizeof(queue->aiocbList[0]))` be more reliable? it'd guarantee the types match.

Reply:
I don't think that's correct. This calloc is allocating an array of pointers to `struct aiocb` not an array of `struct aiocb`

Reply:
I will leave this one in place then, it seems to be the style generally in blktap.

Interestingly Edvins suggestion does compile though, I wonder if the compiler is really opening an uninitalized array or if it is doing some smart type inference during compile time.

Reply:
Actually, it's probably the same thing if queue->aiocblist is a `struct aiocb **`